daprdocs/content/en/developing-applications/openai-agents/openai-agents-sessions.md

```diff
@@ -0,0 +1,119 @@
+
+To initialize the Dapr control plane containers and create a default configuration file, run:
+
+```bash
+dapr init
+```
+
+Verify you have container instances with `daprio/dapr`, `openzipkin/zipkin`, and `redis` images running:
+
+```bash
+docker ps
+```
+
+### Install Python
+
+{{% alert title="Note" color="info" %}}
+Make sure you have Python already installed. `Python >=3.10`. For installation instructions, visit the official [Python installation guide](https://www.python.org/downloads/).
+{{% /alert %}}
+
+### Install Dependencies
+
+```bash
+pip install openai-agents dapr
+```
+
+### Create an OpenAI Agent
+
+Let's create a simple OpenAI agent. Put the following in a file named `openai_agent.py`:
+
+```python
+import asyncio
+from agents import Agent, Runner
+from agents.extensions.memory.dapr_session import DaprSession
+
+async def main():
+    agent = Agent(
+        name="Assistant",
+        instructions="Reply very concisely.",
+    )
+
+    session = DaprSession.from_address(
+        session_id="123",
+        state_store_name="statestore"
+    )
+
+    result = await Runner.run(agent, "What city is the Golden Gate Bridge in?", session=session)
+    print(result.final_output)
+
+    result = await Runner.run(agent, "What state is it in?", session=session)
+    print(result.final_output)
+
+    result = await Runner.run(agent, "What's the population?", session=session)
+    print(result.final_output)
+
+asyncio.run(main())
+```
+
+### Set an OpenAI API key
+
+```bash
+export OPENAI_API_KEY=sk-...
+```
+
+### Create a Python venv
+
+```bash
+python -m venv .venv                                                                                                                                                                      
+source .venv/bin/activate  # On Windows: .venv\Scripts\activate
+```
+
+### Create the database component
+
+The component file is how Dapr connects to your databae. The full list of supported databases can be found [here]({{% ref supported-state-stores %}}). Create a `components` directory and this file in it:
+
+`statestore.yaml`:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+```
+
+### Run The Agent
+
+Now you will run the local Dapr process and your Python script using the Dapr CLI, a local development tool that makes it easy to launch both Dapr and your app side-by-side.
+
+```bash
+dapr run --app-id openaisessions --dapr-grpc-port 50001 --resources-path ./components -- python3 ./openai_agent.py
+```
+
+Open `http://localhost:9411` to view your the traces and dependency graph!
```

Comment:
See [the session data stored in Redis]({{% ref "getting-started/get-started-api" %}}#step-4-see-how-the-state-is-stored-in-redis) with 

```bash
hgetall "123:messages" 
```